### PR TITLE
doc: add a basic release process

### PR DESCRIPTION
**Pull request by Stebalien**
1. A simple checklist.
2. A script to generate release notes.

I'm explicitly ignoring anything like a branching model for the moment. We don't have a long RC process so that shouldn't be required (yet).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  - Create testing branches in upstream repos with the new go-libp2p release and run CI/tests.
+    - [ ] [filecoin-project/lotus](https://github.com/filecoin-project/lotus)
+    - [ ] [ipfs/go-bitswap](https://github.com/ipfs/go-bitswap)
+    - [ ] [ipfs/go-ipfs](https://github.com/ipfs/go-ipfs)
+    - [ ] [libp2p/go-libp2p-kad-dht](https://github.com/libp2p/go-libp2p-kad-dht)
+    - [ ] [libp2p/go-libp2p-pubsub](https://github.com/libp2p/go-libp2p-pubsub)
+  - _(someday)_ Run upstream testground tests. Unfortunately, this is too time consuming at the moment.
+    - _(someday)_ Run bitswap testground tests.
+    - _(someday)_ Run DHT testground tests.
+- [ ] **Stage 2 - Infrastructure Testing**
```
Comment (marten-seemann) [inline .github/ISSUE_TEMPLATE/release.md:33]: Whose responsibility is this? Do we need to get the infrastructure team involved at that point?

Comment (Stebalien) [reply .github/ISSUE_TEMPLATE/release.md:33]: We're responsible for creating the appropriate branches, infra is responsible for deploying.

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## Changelog
+
+< changelog generated by scripts/mkreleaselog >
+
+## ✅ Release Checklist
+
+- [ ] **Stage 0 - Automated Testing**
+    - [ ] Fork a new `release-vX.Y.Z` branch from `master`, freezing master.
+    - [ ] Make sure local tests are passing.
+- [ ] **Stage 1 - Upstream Testing**
```
Comment (vasco-santos) [inline .github/ISSUE_TEMPLATE/release.md:23]: Can we guarantee libp2p interop tests work? That basically includes:
- update `go-libp2p-daemon`
- generate binary for https://github.com/libp2p/npm-go-libp2p-dep
- update interop [dep](https://github.com/libp2p/interop/blob/master/package.json#L42)

I would like to have all these automated, but I think the way to go will be to use the testground for interop in the future

Comment (Stebalien) [reply .github/ISSUE_TEMPLATE/release.md:23]: Ah, yes. SGTM.

Comment (Stebalien) [reply .github/ISSUE_TEMPLATE/release.md:23]: Done.

File: README.md [hunk 3]
```diff
@@ -110,6 +110,10 @@ If you experience any issues migrating from gx to gomod, please [join the discus
 
 `go test ./...` will run all tests in the repo.
 
+### Releasing
+
+Please start a release by opening a new [Libp2p Release](https://github.com/libp2p/go-libp2p/issues/new?assignees=&labels=kind/tracking&template=release.md) issue.
```
Comment (mxinden) [inline README.md:115]: :rocket:

**Review by mxinden — APPROVED** [review]
This pull request looks good to me, thanks for documenting the process.

I am approving as a review was requested from my side. I am not familiar with general Golang release best practices, nor am I familiar with the go-libp2p repository architecture. With that in mind, I don't think my review should be given much weight.

**Comment by Stebalien** [comment]
(merging so I can create a release issue so we can test the process)

**Pull request merged**